_test/test_sqw_class/test_dnd_constructor.m

```diff
@@ -0,0 +1,108 @@
+classdef test_dnd_constructor < TestCase
+
+properties
+    dnd_2d_filename = 'dnd_2d.sqw';
+    dnd_files_path = '../test_sqw_file/';
+
+    test_dnd_2d_fullpath = '';
+end
+
+methods
+
+    function obj = test_dnd_constructor(~)
+        obj = obj@TestCase('test_dnd_constructor');
+
+        test_dnd_2d_file = java.io.File(pwd(), fullfile(obj.dnd_files_path, obj.dnd_2d_filename));
+        obj.test_dnd_2d_fullpath = char(test_dnd_2d_file.getCanonicalPath());
+    end
+
+    function test_d2d_class_follows_expected_class_heirarchy(obj)
+        dnd_obj = d2d();
+
+        assertTrue(isa(dnd_obj, 'd2d'));
+        assertTrue(isa(dnd_obj, 'DnDBase'));;
+        assertTrue(isa(dnd_obj, 'SQWDnDBase'));
+    end
+
+    function test_default_constructor_returns_empty_instance(obj)
+        dnd_obj = d2d();
+
+        assertTrue(isa(dnd_obj, 'd2d'));
+        assertEqual(numel(dnd_obj.pax), 2);
```

Comment:
The name of this test suggests that we should have an `isempty` call in here somewhere. I see that `isempty` isn't implemented on `DnDBase`, should we be checking `isempty(dnd_obj.s) && isempty(dnd_obj.e)` is true?

Reply:
This mirrors the equivalent SQW test: 
```    
function test_default_constructor_returns_empty_instance(obj)
        sqw_obj = sqw();

        assertTrue(isa(sqw_obj, 'sqw'));
        assertEqual(sqw_obj.main_header, struct([]));
        assertEqual(sqw_obj.header, struct([]));
        assertEqual(sqw_obj.detpar, struct([]));
        assertEqual(sqw_obj.data.pix, PixelData());
        assertEqual(numel(sqw_obj.data.pax), 0);
 end
```

There is no isempty method on either class, I'm not sure there any end user use-case for such a function either.